### ISSUE TITLE
setIcon() not working on IconicsImageView

### ISSUE THREAD

**Issue by r748** [mikepenz/Android-Iconics#320]
Hello Mike,

Thanks for this nice library!

Just updated to
```
compile 'com.mikepenz:iconics-core:2.9.1@aar'
compile 'com.mikepenz:iconics-views:2.9.2@aar'
compile 'com.mikepenz:google-material-typeface:3.0.1.1.original@aar'
```

I'm using your FastAdapter library. The AbstractItem has three IconicsImageViews in it's layout resource. Two of these icons are defined in xml and show up correctly in the RecyclerView. However on one IconicsImageView I set the icon dynamically the onBind method. But with the newest version of Iconics library, setIcon(IIcon) is not working for me, the icon doesn't show up.

The icon's xml looks like this

```
<com.mikepenz.iconics.view.IconicsImageView
android:id="@+id/icon"
android:layout_width="wrap_content"
android:layout_height="wrap_content"
app:iiv_color="@color/white"
app:iiv_size="28dp" />
```

And I'm using `viewHolder.icon.setIcon(icon);` whereas icon is an IIcon.

Any ideas?

Thanks and best regards,
Rob

**Comment by mikepenz** [comment]
The icon you set does it have a size or anyhting? 

Seems weird why it wouldn't work. I directly set the drawable:
https://github.com/mikepenz/Android-Iconics/blob/develop/iconics-view-library/src/main/java/com/mikepenz/iconics/view/IconicsImageView.java#L88

**Comment by r748** [comment]
Hi Mike,

Thanks for your quick reply.

In the meanwhile, I figured out that the icons show up, if I set the icon with all it's attributes in the `onBind()` method like this:
```
IconicsDrawable drawable = new IconicsDrawable(context, icon).sizeDp(28).color(Color.WHITE);
viewHolder.icon.setIcon(drawable);
```

But in the previous version of Iconics library I was using, with `setIcon()` only the icon drawable was changed, but the attributes defined in xml (e.g. iiv_size, iiv_color) were preserved.

**Comment by mikepenz** [comment]
@r748 ah ok I see. I'll have a look

**Comment by r748** [comment]
Thanks, Mike!

Just for info:
The last Iconics version I was using, where `setIcon()` only changed the icon and xml attributes were preserved, was 2.8.5. Also, I upgraded compileSdkVersion from 25 to 26 at the same time, as I upgraded Iconics library. So possibly this may be a bug in the `setImageDrawable()` method of SDK 26?!

**Comment by mikepenz** [comment]
@r748 no I am sure the support libs 26 makes no difference. I'll look into it as soon as I have time

**Comment by r748** [comment]
Thank you very much!

**Comment by r748** [comment]
Just discovered, `setColor()` on `IconicsImageView` in `onBind()` method also seems to be broken...

**Comment by mikepenz** [comment]
@r748 the `setColor()` should work. at least it does in the sample

**Comment by mikepenz** [comment]
@r748 could you please try if this: https://github.com/mikepenz/Android-Iconics/commit/7d8bdd98fe1b1aa4e0bf8914e89444730dfaf815 resolves your issues?

**Comment by r748** [comment]
Hey Mike,

Your changes resolve my all issues concerning `setIcon()`. Thank you very much!

However one strange issue remains with `setColor()`: If I set an icon's color in xml (`app:iiv_color`) and then try to change it with `setColor()`, the icon color is not always set to the new color. It seems that if the old color has a certain percentage of transparency (e.g. #12000000), the new color inherits the transparency. Is this intentionally?

Cheers!
Rob

**Comment by mikepenz** [comment]
@r748 yeah it is somewhat intentionally

see the following in the `IconcisDrawable`

```java
int alpha = Color.alpha(color);
if (alpha != 255 && alpha != mAlpha) {
    setAlpha(alpha);
}
```

basically I want to allow setting the alpha independent of the color, it's hard to detect if the new color should change the pre-set alpha.

**Comment by mikepenz** [comment]
released with v2.9.3

**Comment by r748** [comment]
Thanks, Mike!
